src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -359,4 +360,95 @@ public void registerUserForCourse(User user, Course course) {
         auditEventRepository.add(auditEvent);
         log.info("User " + user.getLogin() + " has successfully registered for course " + course.getTitle());
     }
+
+    /**
+     * Get the active students for this particular course
+     *
+     * @param courseId the id of the course
+     * @param periodIndex an index indicating which time period, 0 is current week, -1 is one week in the past, -2 is two weeks in the past ...
+     * @return An Integer array containing active students for each index
+     */
+    public Integer[] getCourseStatistics(Long courseId, Integer periodIndex) {
+        ZonedDateTime now = ZonedDateTime.now();
+
+        LocalDateTime localStartDate = now.toLocalDateTime().with(DayOfWeek.MONDAY);
+        LocalDateTime localEndDate = now.toLocalDateTime().with(DayOfWeek.SUNDAY);
+        ZoneId zone = now.getZone();
+        ZonedDateTime startDate = localStartDate.atZone(zone).minusWeeks(3 + (4 * (-periodIndex))).withHour(0).withMinute(0).withSecond(0).withNano(0);
+        ZonedDateTime endDate = periodIndex != 0 ? localEndDate.atZone(zone).minusWeeks(4 * (-periodIndex)).withHour(23).withMinute(59).withSecond(59)
+                : localEndDate.atZone(zone).withHour(23).withMinute(59).withSecond(59);
+        List<Map<String, Object>> outcome = courseRepository.getCourseStatistics(courseId, startDate, endDate);
+        List<Map<String, Object>> distinctOutcome = removeDuplicatesFromMapList(outcome, startDate);
+        return createResultArray(distinctOutcome, endDate);
+    }
+
+    /**
+     * the List of maps result contains duplicated entries. This method compares the values and returns a List<Map<String, Object>>
+     * without duplicated entries
+     *
+     * @param result the result given by the Repository call
```

Comment:
It was not so easy to understand how the method does what it does. Maybe renaming `result` to something more descriptive and explaining what it exactly is could help. Is it a List of Maps of users to their active period?